projects/online/online/main.py

```diff
@@ -122,19 +123,16 @@ def search(
 
     state = snapshotter.initial_state
     for X, t0, ready in data_it:
-        # TODO:
-        # here we can handle any subprocess
-        # errors - I think at the least we
-        # need to send an email that
-        # something failed and the pipeline
-        # needs to be restarted
+        # handle any subprocess
         try:
             name, error, tb = error_queue.get_nowait()
         except Empty:
             pass
         else:
             logging.error(f"Error in subprocess {name}: {str(error)}")
             logging.error(tb)
+            if emails is not None:
```

Comment:
I think it would be good to have some extra safety measures to make sure we don't accidentally spam emails out when testing the pipeline. Maybe have a flag that needs to be set in addition to the email list, or add a logging statement about which emails error messages will be sent to.

Reply:
The flag would just be to comment out the emails since they're optional. Definitely will add logging statement up front about whether emails are being sent or not, and if so who is receiving them

Reply:
Yeah, I was thinking like as a redundancy. But I think logging will be sufficient